tests/cli/deployment/test_deployment_run.py

```diff
@@ -0,0 +1,318 @@
+    ],
+)
+def test_start_at_option_invalid_input(
+    deployment_name: str, start_at: str, expected_output: str
+):
+    invoke_and_assert(
+        command=[
+            "deployment",
+            "run",
+            deployment_name,
+            "--start-at",
+            start_at,
+        ],
+        expected_code=1,
+        expected_output=expected_output,
+    )
+
+
+@pytest.mark.parametrize(
+    "start_at,expected_start_time",
+    [
+        ("12-1-2022 5pm", "2022-12-01 17:00:00"),
+        ("1/1/30", "2030-01-01 00:00:00"),
+        ("13/1/2020 13:31", "2020-01-13 13:31:00"),
+        ("9pm December 31st 2022", "2022-12-31 21:00:00"),
+        ("January 2nd 2023", "2023-01-02 00:00:00"),
+    ],
+)
+async def test_start_at_option_displays_scheduled_start_time(
+    deployment_name: str,
+    start_at: str,
+    expected_start_time: str,
+):
+    await run_sync_in_worker_thread(
+        invoke_and_assert,
+        command=[
+            "deployment",
+            "run",
+            deployment_name,
+            "--start-at",
+            start_at,
+        ],
+        expected_output_contains=["Scheduled start time:", expected_start_time],
+    )
+
+
+@pytest.mark.parametrize(
+    "start_at,expected_start_time",
+    [
+        ("5-17-23 5:30pm UTC", pendulum.parse("2023-05-17T17:30:00")),
+        ("5-20-2020 5:30pm EDT", pendulum.parse("2020-05-20T17:30:00", tz="EST5EDT")),
+        ("01/31/23 5:30 CST", pendulum.parse("2023-01-31T05:30:00", tz="CST6CDT")),
+        ("5-20-23 5:30pm PDT", pendulum.parse("2023-05-20T17:30:00", tz="PST8PDT")),
+        ("01/31/23 5:30 PST", pendulum.parse("2023-01-31T05:30:00", tz="PST8PDT")),
+    ],
+)
+async def test_start_at_option_with_tz_displays_scheduled_start_time(
+    deployment_name: str, start_at: str, expected_start_time: DateTime
+):
+    expected_start_time_local = expected_start_time.in_tz(pendulum.tz.local_timezone())
+    expected_display = (
+        expected_start_time_local.to_datetime_string()
+        + " "
+        + expected_start_time_local.tzname()
+    )
+
+    await run_sync_in_worker_thread(
+        invoke_and_assert,
+        command=[
+            "deployment",
+            "run",
+            deployment_name,
+            "--start-at",
+            start_at,
+        ],
+        expected_output_contains=["Scheduled start time:", expected_display],
+    )
+
+
+@pytest.mark.parametrize(
+    "start_at,expected_start_time",
+    [
+        (
+            "12/20/2022 1am",
+            DateTime(2022, 12, 20, 1, 0, 0, tzinfo=pendulum.tz.local_timezone()),
+        ),
+        (
+            "1-1-2020",
+            DateTime(2020, 1, 1, 0, 0, 0, tzinfo=pendulum.tz.local_timezone()),
+        ),
+        (
+            "5 June 2015",
+            DateTime(2015, 6, 5, 0, 0, 0, tzinfo=pendulum.tz.local_timezone()),
+        ),
+    ],
+)
+async def test_start_at_option_schedules_flow_run_in_future(
```

Comment:
Might want to rename this test since some of these are not scheduled in the future